File: backend/naomie/src/main/scala/com/github/GuillaumeReinisch/Naomie/models/Graphic.scala
package com.github.GuillaumeReinisch.Naomie.models

/**
  * Created by Guillaume on 17/02/17.
  */
case class Graphic( uid: String, name: String , axis: List[String], metadata : Map[String,String] = Map.empty[String,String] , overlays: List[String] = List.empty[String]) {

}

case class GraphicCollection( uid: String, name: String , userName: String,creationDate: String, description: String, graphics: List[String] = List.empty[String]) {
}

File: frontend/naomie1/src/app/models/graphic.ts
export class Axis {
  constructor(public name: string, public formula:  string){}
}

export class Variable {
  constructor(public displayName: string, public name:  string){}
}

export class Graphic {
  //val name: String , val axis: List[String], val metadata : Map[String,String] = Map[String,String]()
  name: string;
  uid: string;
  axis: string[];
  metadata: Map<string,string>;
}

export class GraphicCollectionForm {
  constructor(public name:string, public user:string, public description : string){}
}
export class GraphicCollection {
  constructor(public name:string, public key:string,public user:string, public description : string){}
}
